src/pages/iv-opening-cost/Forms/OpeningCostForm.js

```diff
@@ -0,0 +1,163 @@
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      year: yup.number().required(),
+      itemId: yup.string().required(),
+      avgCost: yup.number().min(0).max(999999999)
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: InventoryRepository.OpeningCost.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+
+        formik.setValues({
+          ...obj,
+          recordId: String(obj.year) + String(obj.itemId)
+        })
+      } else toast.success(platformLabels.Edited)
+
+      invalidate()
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      if (record?.itemId && record?.year && recordId) {
+        const res = await getRequest({
+          extension: InventoryRepository.OpeningCost.get,
+          parameters: `_itemId=${record?.itemId}&_fiscalYear=${record?.year}`
+        })
+
+        formik.setValues({
+          ...res.record,
+          recordId: String(res.record.year) + String(res.record.itemId)
+        })
+      }
+    })()
+  }, [])
+
+  return (
+    <FormShell 
+      resourceId={ResourceIds.InventoryOpeningCosts} 
+      form={formik} 
+      maxAccess={maxAccess} 
+      editMode={editMode}
+      infoVisible={false}
+      isSavedClear={false}
+      isInfo={false}
+      isCleared={false}
+    >
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={SystemRepository.FiscalYears.qry}
+                readOnly={editMode}
+                name='year'
+                label={labels?.fiscalYear}
+                valueField='fiscalYear'
+                displayField='fiscalYear'
+                values={formik.values}
+                required
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('year', newValue?.fiscalYear)
+                }}
+                error={formik.touched.year && Boolean(formik.errors.year)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <ResourceLookup
+                endpointId={InventoryRepository.Item.snapshot}
+                name='itemId'
+                label={labels?.sku}
+                readOnly={editMode}
+                valueField='recordId'
+                displayField='sku'
+                valueShow='sku'
+                secondDisplayField={false}
+                form={formik}
+                columnsInDropDown={[
+                  { key: 'sku', value: 'SKU' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('itemId', newValue?.recordId)
+                  formik.setFieldValue('itemName', newValue?.name)
```

Comment:
formik does not have itemName in the initial values should't it at lease be a null value?